mozilla-services/socorro
# bug 1553314: remove configuration from Rules

This adjusts the `ProcessorPipeline` to own configuration of the rules in the
pipeline. Configuration is now defined on that class and it passes in necessary
data through init args to the rules that require it.  In this way, rules are
just transformation classes and aren't configman-style classes.

This adds logging of the pipeline rules and their configuration. This will
make it a lot easier to notice and debug rules problems in the future.

Doing this also fixed configuration for the `JitCrashCategorizeRule` which was
totally busted and probably has been for some time. Now rule configuration is
namespaced which allows us to configure both the `BreakpadStackwalkerRule2015`
and `JitCrashCategorizeRule`.

This also removed all the confusing duplicated (and triplicated in some places)
configuration which was confusing and causing problems.

I also nixed the `ExternalProcessRule` and folded it into the rules that subclassed it. Those three rules were hard to follow because of all the back and forth between super/subclasses and selective overriding. It was way easier to just break them up and move the shared bits to a function.

To test:

1. run tests
2. process some crashes
   1. does the rule output on startup look sane?
   2. did BreakpadStackwalkerRule2015 run? do crashes get signatures?
   3. did the BetaVersionRule run? you need to make sure to process a Firefox beta channel crash to check this.

Note that the `JitCrashCategorizeRule` isn't quite fixed, yet. It still won't do anything because it's kicking off before signature generation, but depends on the signature to run. I want to fix that in a follow-up PR because it hasn't run in so long that I don't know what it's supposed to do or whether it works at all.
